vllm/utils.py

```diff
@@ -2253,3 +2253,39 @@ def import_pynvml():
     """
     import vllm.third_party.pynvml as pynvml
     return pynvml
+
+
+def warn_for_unimplemented_methods(cls: Type[T]) -> Type[T]:
+
+    original_init = cls.__init__
+
+    def find_unimplemented_methods(self: object):
+        unimplemented_methods = []
+        for attr_name in dir(self):
+            # bypass inner method
+            if attr_name.startswith('_'):
+                continue
+            base_method = getattr(self, attr_name)
+            # get the func of callable method
+            if callable(base_method):
+                base_method_name = base_method.__func__
+            else:
+                continue
+            class_method_name = getattr(cls, attr_name, False)
```

Comment:
it seems it is still not checking `NotImplementedError`?

Reply:
whoops, my bad, please check updated commit